### PR TITLE
oci sources integration

### PR DESCRIPTION
**Pull request by lcouzens**
Fixes [COST-2392](https://issues.redhat.com/browse/COST-2392)

Changes proposed in this PR:
* Adding OCI pieces to sources flow

Testing Instructions:
1. Might need to have sources running locally to test this :grimacing: 

---
Additional Context:
Reach out to Luke/Michael if you want instructions for how to run sources locally.

### CODE REVIEW

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
2e3aafd0-6355-4636-b2cf-d51b3ea3da3e
ce964484-0617-4296-b7d9-fb2605db2552

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
b9da7432-214b-4a2b-bab3-8748426ecfd5
f84f3335-84de-4519-864b-8238735d2816

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
6c10f4ae-9a93-4faa-8232-1ec9c8ddeb1f
8b23be23-debb-4b1c-a4a6-f1b77eede48e

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
13e3bb39-313c-4aa3-bb93-56a8005f9363
75f4fcc6-a397-49f6-a9d2-9950fad532bd

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
7b0f7271-136a-4666-a101-71f3947395ba
8525c00b-27a9-499e-930d-c36b072aad61

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
1d3e0b88-cab5-41c9-a53e-28f7bf48d9e1
5f06e12a-2716-469a-a724-7074aaf1b231

**Comment by lcouzens** [comment]
`16:47:42 FAILED tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_endpoint_date_range[previous-month-yesterday-monthly]
16:47:42 FAILED tests/rest_api/v1/test__ocp_compute_reports.py::test_api_ocp_compute_endpoint_date_range[previous-month-yesterday-daily]
16:47:42 FAILED tests/rest_api/v1/test_source.py::test_api_gcp_source_crud`

File: koku/sources/kafka_message_processor.py [hunk 1]
```diff
@@ -152,6 +156,11 @@ def save_credentials(self):
             LOG.info(f"[save_credentials] source_type not found for source_id: {self.source_id}")
             return
 
+        if source_type == Provider.PROVIDER_OCI or source_type == Provider.PROVIDER_OCI_LOCAL:
+            # OCI sources do not have authentication, so skip running thru this function
+            LOG.info("[save_credentials] skipping for OCI source")
+            return {}
```
Comment (Red-HAP) [inline koku/sources/kafka_message_processor.py:162]: **question:** This method already has two return types `None` and `bool` Is it your intention to add a _third_ return type of `dict`?

File: koku/sources/test/test_kafka_message_processor.py [hunk 2]
```diff
@@ -366,6 +380,16 @@ def test_save_sources_details_unknown_source_type(self):
...
                         mock_details_save.assert_not_called()
 
+    def test_save_oci_credentials(self):
+        """Test save oci credentials calls add_provider_sources_auth_info."""
+        event = choice(EVENT_LIST)
+        msg = msg_generator(event)
+        provider = "OCI"
+        processor = KafkaMessageProcessor(msg, event, COST_MGMT_APP_TYPE_ID)
+        with patch("sources.storage.get_source_type", return_value=provider):
+            result = processor.save_credentials()
```
Comment (Red-HAP) [inline koku/sources/test/test_kafka_message_processor.py:390]: **question:** Is `dict` really needed here? Would `True` work just as well?

File: koku/sources/test/test_storage.py [hunk 3]
```diff
@@ -317,6 +317,38 @@ def test_screen_and_build_provider_sync_create_event(self):
                 ),
                 "expected_response": {},
             },
+            {
```
Comment (Red-HAP) [inline koku/sources/test/test_storage.py:320]: **nitpick (non-blocking):** keyword args would sure help here.

**Review by Red-HAP — COMMENTED** [review]
I'd like the questions around a third return type addressed with a clear response or correction to make it in-line with the rest of the provider types.

**Review by Red-HAP — APPROVED** [review]

**Pull request merged**